packages/components/src/components/scroll-buttons/scroll-buttons.tsx

```diff
@@ -5,12 +5,32 @@ import { Component, h, JSX, Method, Prop, Watch } from '@stencil/core';
     styleUrl: 'scroll-buttons.scss',
     shadow: true
 })
+
 export class ScrollButtons {
     /**
      * Reference to the element with scroll on parent element.
+     *
      * @type {HTMLDivElement}
      */
     @Prop() scrollContainerElementRef: HTMLDivElement;
+
+    /**
+     * Locations visible when specific category is selected.
+     */
+    @Prop() locations;
+
+    /**
+     * Method.
+     */
+    @Method()
+    public async newCategorySelected(locations): Promise<any> {
+        this.locations = locations;
+        this.updateScrollButtonsState();
+    }
```

Comment:
The scroll buttons element is not tightly coupled with Locations and Categories, but can control any type of content. Please make the naming more generic.

Reply:
✅